demo/demo.html

```diff
@@ -6,26 +6,29 @@
         <title>demo</title>
         <meta name="description" content="">
         <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
-		<script>
-			var app = {};
-			app.configuration = { data: {} };
-			app.configuration.lang = 'en';
-			app.configuration.data.page = "test";
-		</script>
-    </head>
-    <body>
-        <p>Hello world! This is HTML5 Boilerplate.</p>
-        <script src="../configurationLoader.js"></script>
+        <script src="../dist/biotopeConfiguration.js"></script>
         <script>
-			configurationLoader({
-				json: 'demo.json',
-				data: app.configuration,
-				overwriteInlineConfigs: false
-			}).then(function(configuration) {
-				console.log(configuration.get('lang')); //depending on whether overwriteInlineConfigs flag is true or false this will return "de" or "en"
+            myApp = {};
+            myApp.config = biotopeConfiguration();
+            myApp.config.set('data.language', 'en');
+
+            console.log(myApp.config.get('data.language'));
 
-				console.log(configuration.get('data')); //this would always return properties "id" and "page", value of "page" property depends on overwriteInlineConfigs flag.
-			});
+            myApp.config.setRemote({
+                url: 'demo.json',
+				overwrite: false
+            }).then(function() {
+                //myApp.config is now updated with data from json file.
+                myApp.init();
+            });
+
+            myApp.init = function() {
+                console.log('Configuration is ready!', myApp.config.get('data'));
```

Comment:
i still think this is prune to race conditions. in most cases this line will be in our main.js which is loaded after the DOM and all other scripts so i see a chance that the promise is ready before the function call myApp.init is available.

Reply:
I updated demo to use the example we had in readme.md. This is a bulletproof way to run myApp.init() since now it only executes when both DOM and configuration are ready.